### PR TITLE
Hud documentation

### PR DESCRIPTION
**Pull request by KarolDenst**

### CODE REVIEW

File: Hud/Shaders/HudShader.cs [hunk 1]
```diff
@@ -19,18 +25,34 @@ public static HudShader Create()
         return new HudShader(shader);
     }
 
-    public void SetProjection(Matrix4 projection) => SetMatrix4("projection", projection);
+    private void SetProjection(Matrix4 projection) => SetMatrix4("projection", projection);
 
+    /// <summary>
+    /// Sets up the shader.
+    /// </summary>
+    /// <param name="aspectRatio"></param>
```
Comment (AleksyBalazinski) [inline Hud/Shaders/HudShader.cs:33]: missing parameters descriptions

File: Hud/Sprites/SpriteSheetMetadata.cs [hunk 2]
```diff
@@ -2,19 +2,34 @@
...
+/// Represents a sprite sheet metadata.
+/// </summary>
 [Serializable]
 public class SpriteSheetMetadata
 {
+    /// <summary>
+    /// Width of the sprite sheet.
+    /// </summary>
     [JsonPropertyName("width")]
     public int Width { get; set; }
```
Comment (AleksyBalazinski) [inline Hud/Sprites/SpriteSheetMetadata.cs:15]: The properties in this class could be private fields, but I'm not sure if that would work with the serializer. If it'd be possible to make the private we could do that I reckon

File: Hud/Sprites/SpriteSheetMetadata.cs [hunk 3]
```diff
@@ -2,19 +2,34 @@
...
+    /// <summary>
+    /// Items in the sprite sheet.
+    /// </summary>
     [JsonPropertyName("items")]
     public List<Sprite> Items { get; set; } = null!;
 
+    /// <summary>
+    /// Gets the rectangles of the sprites in the sprite sheet. Rectangles describe the position of a sprite in the sprite sheet.
+    /// </summary>
+    /// <returns></returns>
```
Comment (AleksyBalazinski) [inline Hud/Sprites/SpriteSheetMetadata.cs:32]: This is a rather complex return type, consider describing the elements of the tuple

File: Hud/Widgets/Colors/ColorGetter.cs [hunk 4]
```diff
@@ -6,6 +6,11 @@ namespace Hud.Widgets.Colors;
 /// </summary>
 public static class ColorGetter
 {
+    /// <summary>
+    /// Gets the Vector4 representation of the Color.
```
Comment (AleksyBalazinski) [inline Hud/Widgets/Colors/ColorGetter.cs:10]: `Vector4` -> `<see cref="Vector4"/>`

File: Hud/Widgets/Colors/ColorGetter.cs [hunk 5]
```diff
@@ -6,6 +6,11 @@ namespace Hud.Widgets.Colors;
...
+    /// </summary>
+    /// <param name="color">The color.</param>
+    /// <returns>The Vector4 representation of the Color.</returns>
```
Comment (AleksyBalazinski) [inline Hud/Widgets/Colors/ColorGetter.cs:13]: same & color w/o 1st letter capital

File: Hud/Widgets/Context.cs [hunk 6]
```diff
@@ -11,10 +11,21 @@ namespace Hud.Widgets;
 /// <param name="Size">Width and height of the Context</param>
 public record Context(HudShader Shader, Vector2 Position, Vector2 Size)
 {
+    /// <summary>
```
Comment (AleksyBalazinski) [inline Hud/Widgets/Context.cs:14]: context w/o 1st letter capital

File: Hud/Widgets/Context.cs [hunk 7]
```diff
@@ -11,10 +11,21 @@ namespace Hud.Widgets;
...
+    /// Creates a new Context with the same shader and size but a different position.
+    /// </summary>
+    /// <param name="oldContext">The old context.</param>
+    /// <param name="newPosition">The new position.</param>
     public Context(Context oldContext, Vector2 newPosition) :
         this(oldContext.Shader, newPosition, oldContext.Size)
     { }
 
+    /// <summary>
+    /// Creates a new Context with the same shader but a position and a different size.
```
Comment (AleksyBalazinski) [inline Hud/Widgets/Context.cs:24]: same

File: Hud/Widgets/NoChildren/InputText.cs [hunk 8]
```diff
@@ -28,6 +31,14 @@ internal class InputText : Widget
 
     private readonly string _placeholder = "";
 
+    /// <summary>
+    /// Creates an instance of InputText class.
```
Comment (AleksyBalazinski) [inline Hud/Widgets/NoChildren/InputText.cs:35]: <see cref> stuff

File: Hud/Sprites/SpriteSheetMetadata.cs (hunk 2 above)
Comment (KarolDenst) [reply Hud/Sprites/SpriteSheetMetadata.cs:15]: It fails with the serializer

File: Hud/Shaders/HudShader.cs [hunk 9]
```diff
@@ -19,18 +25,34 @@ public static HudShader Create()
...
     {
         Use();
         var projection = Matrix4.CreateOrthographic(aspectRatio, 1, -1.0f, 1.0f);
         SetProjection(projection);
     }
 
+    /// <summary>
+    /// Sets the color of the rendered object.
+    /// </summary>
+    /// <param name="color"></param>
```
Comment (AleksyBalazinski) [inline Hud/Shaders/HudShader.cs:44]: missing params

**Review by AleksyBalazinski — APPROVED** [review]

**Pull request merged**